src/core/memory.d

```diff
@@ -919,3 +919,184 @@ extern (C) private pure @system @nogc nothrow
 
     pragma(mangle, "free") void fakePureFree(void* ptr);
 }
+
+private extern (C) void _d_delinterface(void** p);
+private extern (C) void _d_delclass(Object* p);
+private extern (C) void _d_delstruct(void** p, TypeInfo_Struct inf);
+
+/**
+The `delete` keyword allowed to free GC-allocated memory.
+As this is inherently not `@safe`, it has been deprecated.
+This function has been added to provide an easy transition from `delete`.
+It performs the same functionality as `delete`.
+
+$(REF destroy, object) is `@safe` and should be used if possible.
```

Comment:
Again, I think this should have some information explaining that users should first look to `destroy` and only to `__delete` when `destroy` is not an option.